htlcswitch/link.go

```diff
@@ -1064,6 +1071,51 @@ out:
 	}
 }
 
+// processHodlEvent applies a received hodl event to the provided htlc. When
+// this function returns without an error, the commit tx should be updated.
+func (l *channelLink) processHodlEvent(htlcs []hodlHtlc,
+	hodlEvent invoices.HodlEvent) error {
+
+	hash := hodlEvent.Hash
+	if hodlEvent.Preimage == nil {
```

Comment:
would it be more clear to actually define a event type field?

Reply:
I think I am fine with this. `Hash` is already redundant if `Preimage` is set, and this would add another redundant field.